src/commands/app/install.js

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const BaseCommand = require('../../BaseCommand')
+const { Flags } = require('@oclif/core')
+const aioLogger = require('@adobe/aio-lib-core-logging')('@adobe/aio-cli-plugin-app:install', { provider: 'debug' })
+const path = require('node:path')
+const fs = require('fs-extra')
+const unzipper = require('unzipper')
+const execa = require('execa')
+const { validateJsonWithSchema } = require('../../lib/install-helper')
+
+class InstallCommand extends BaseCommand {
+  async run () {
+    const { args, flags } = await this.parse(InstallCommand)
+
+    aioLogger.debug(`flags: ${JSON.stringify(flags, null, 2)}`)
+    aioLogger.debug(`args: ${JSON.stringify(args, null, 2)}`)
+
+    // resolve to absolute path before any chdir
+    args.path = path.resolve(args.path)
+    aioLogger.debug(`args.path (resolved): ${args.path}`)
+
+    let outputPath = flags.output
+    // change the cwd if necessary
+    if (outputPath !== '.') {
+      outputPath = path.resolve(flags.output)
+      // TODO: confirm if dir exists for overwrite
+      await fs.ensureDir(outputPath)
+      process.chdir(outputPath)
+      aioLogger.debug(`changed current working directory to: ${outputPath}`)
+    }
+
+    await this.validateZipDirectoryStructure(args.path)
+    await this.unzipFile(args.path, outputPath)
+    await this.validateConfig(outputPath, 'app.config.yaml')
+    await this.validateConfig(outputPath, 'deploy.yaml')
+    await this.runTests(outputPath)
+  }
+
+  diffArray (left, right) {
+    if (!left || !right) {
```

Comment:
Nitpick, as IIUC left, is the expected files, but if `left` is `undefined` then this returns `undefined` (meaning `left.length` will throw).
(not sure but maybe we can rename left to expected, and default it to `[]` to make it future proof)

Reply:
you're right, the difference should always be at least "zero" i.e the empty array []. will make the recommended changes

Reply:
fixed in https://github.com/adobe/aio-cli-plugin-app/pull/662/commits/be2ca51acb754b606cec72ba70a12daacf386d24

Reply:
thanks lgtm